frontend/src/libs/components/dropdown-menu/dropdown-menu.tsx

```diff
@@ -0,0 +1,62 @@
+import { Link } from '#libs/components/components.js';
+import { getValidClassNames } from '#libs/helpers/helpers.js';
+import { useCallback, useLocation, useState } from '#libs/hooks/hooks.js';
+import { type Route } from '#libs/types/types.js';
+
+import styles from './styles.module.scss';
+
+type Properties = {
+  routes: Route[];
+};
+
+const DropdownMenu: React.FC<Properties> = ({ routes }) => {
+  const { pathname } = useLocation();
+  const [isOpen, setOpen] = useState(false);
+
+  const handleDropdownToggle = useCallback((): void => {
+    setOpen(!isOpen);
+  }, [isOpen]);
+
+  return (
+    <div className={styles['dropdown']}>
+      <button
+        className={styles['dropdown-header']}
+        onClick={handleDropdownToggle}
+      >
+        <span className={styles['bar']} />
+        <span className={styles['bar']} />
+        <span className={styles['bar']} />
```

Comment:
according to the **B4** rule, we can reduce the number of spans and use pseudo-elements instead